models/dojo.py

```diff
@@ -0,0 +1,277 @@
+                person.first_name, assigned_office_space.room_name))
+
+        if person.wants_accomodation.upper() == "Y":
+            available_living_spaces = self.get_available_room_spaces(
+                self.living_rooms)
+            if available_living_spaces == []:
+                print("There is currently no living space")
+            else:
+                assigned_living_space = random.choice(available_living_spaces)
+                assigned_living_space.occupants.append(person)
+                person.living_space_allocated = assigned_living_space.room_name
+                print("{0} has been allocated the livingspace {1}".format(
+                    person.first_name, assigned_living_space.room_name))
+
+    def print_room(self, arg):
+        """It prints the names of occupantsstated room name."""
+        room_name = arg["<room_name>"]
+        if room_name in self.office_rooms:
+            office_room_info = self.office_rooms[room_name]
+            if office_room_info.occupants == []:
+                raise Exception("Office room is empty")
+            else:
+                print('Office room occupants --- {}'.format(office_room_info.occupants))
+
+        elif room_name in self.living_rooms:
+            living_room_info = self.living_rooms[room_name]
+            if living_room_info.occupants == []:
+                raise Exception("Living room is empty")
+            else:
+                print('Living room occupants --- {}'.format(living_room_info.occupants))
+        else:
+            raise Exception("The room has not been created")
+
+    def print_allocations(self, arg):
+        """The function prints a list of all
+         allocated rooms with their members."""
+        # pdb.set_trace()
+        allocation_file = arg["--o"]
+
+        for office_name, office_info in self.office_rooms.items():
+            if len(office_info.occupants) != 0:
+                if allocation_file is None:
+
+                    print(
+                        office_info.room_name + self.divider
+                        + str(office_info.occupants) + "\n")
+                else:
+                    result = open(allocation_file + ".txt", "w")
+                    result.write(
+                        office_info.room_name + self.divider
+                        + str(office_info.occupants))
+                    result.close()
+
+        for living_name, living_info in self.living_rooms.items():
+            if len(living_info.occupants) != 0:
+                if allocation_file is None:
+                    print(
+                        living_info.room_name + self.divider
+                        + str(living_info.occupants) + "\n")
+                else:
+                    result = open(allocation_file + ".txt", "w")
+                    result.write(
+                        living_info.room_name + self.divider
+                        + str(living_info.occupants))
+                    result.close()
+
```

Comment:
line 128-140 and lines 143-153 are looking very similar, could you refactor to make more DRY?